### PR TITLE
Esil interrupts

### PR DESCRIPTION
**Pull request by condret**
== fix should be merged befor this, this branch is based on these, because I needed cleaner esil for my talk

### CODE REVIEW

**Comment by condret** [comment]
travis might be red, bc it depends on merging of my test changes for ==

File: libr/anal/esil_interrupt.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		intr->handler->fini (intr->user);
+	}
+	free (intr);
+}
+
+R_API void r_anal_esil_interrupts_init(RAnalEsil *esil) {
+	if (!esil) {
+		return;
+	}
+	esil->interrupts = dict_new (sizeof(ut32), NULL);
```
Comment (radare) [inline libr/anal/esil_interrupt.c]: why use dict? also add space before (

**Comment by radare** [comment]
We should split this pr in 2, and merge your esil refactor asap like we decided after r2con

**Comment by condret** [comment]
yep, i split it

**Comment by radare** [comment]
please rebase. the == pr has been merged

**Pull request closed without merging**